src/utils/viewer/viewer.cpp

```diff
@@ -359,7 +372,43 @@ void Viewer::drawEvent() {
   int DEFAULT_SCENE = 0;
   int sceneID = sceneID_[DEFAULT_SCENE];
   auto& sceneGraph = sceneManager_.getSceneGraph(sceneID);
-  renderCamera_->getMagnumCamera().draw(sceneGraph.getDrawables());
+
+  // XXX
+  auto& drawables = sceneGraph.getDrawables();
+  auto total = drawables.size();
+  size_t visibles = 0;
+  if (enableFrustumCulling_) {
+    Magnum::SceneGraph::DrawableGroup3D visibleDrawables;
+
+    for (int iDrawable = total - 1; iDrawable >= 0; iDrawable--) {
+      // This, of course, will cause segfault.
+      // for (int iDrawable = 0; iDrawable < total; ++iDrawable)
+      scene::SceneNode& node =
+          dynamic_cast<scene::SceneNode&>(drawables[iDrawable].object());
+      auto& aabb = node.getMeshBB();
+      Magnum::Math::Vector4<float> bbMin{aabb.min(), 1.0};
+      Magnum::Math::Vector4<float> bbMax{aabb.max(), 1.0};
+      auto modelview = renderCamera_->getMagnumCamera().cameraMatrix() *
+                       node.absoluteTransformation();
+      if (Magnum::Math::Intersection::rangeFrustum(
+              Magnum::Math::Range3D<float>{(modelview * bbMin).xyz(),
```

Comment:
It transforms the AABB from model frame to the camera frame.
Then the AABB is no longer an AABB, it is an OBB.
So cannot use rangeFrustum to do the culling any more.

Reply:
You'd need to transform the frustum instead to keep AABBs as AABBs -- and I see you did exactly that :)